homeassistant/components/surepetcare/sensor.py

```diff
@@ -30,105 +26,180 @@
     if discovery_info is None:
         return
 
+    entities = []
+
     spc = hass.data[DATA_SURE_PETCARE][SPC]
-    async_add_entities(
-        [
-            FlapBattery(entity[CONF_ID], entity[CONF_NAME], spc)
-            for entity in spc.ids
-            if entity[CONF_TYPE] == SureThingID.FLAP.name
-        ],
-        True,
-    )
 
+    for entity in spc.ids:
+        sure_type = entity[CONF_TYPE]
 
-class FlapBattery(Entity):
-    """Sure Petcare Flap."""
+        if sure_type in [
+            SureProductID.CAT_FLAP,
+            SureProductID.PET_FLAP,
+            SureProductID.FEEDER,
+        ]:
+            entities.append(SureBattery(entity[CONF_ID], sure_type, spc))
+
+        if sure_type in [
+            SureProductID.CAT_FLAP,
+            SureProductID.PET_FLAP,
+        ]:
+            entities.append(Flap(entity[CONF_ID], sure_type, spc))
+
+    async_add_entities(entities, True)
+
+
+class SurePetcareSensor(Entity):
+    """A binary sensor implementation for Sure Petcare Entities."""
+
+    def __init__(
+        self: Entity, _id: int, sure_type: SureProductID, spc: SurePetcareAPI,
+    ):
+        """Initialize a Sure Petcare sensor."""
 
-    def __init__(self, _id: int, name: str, spc):
-        """Initialize a Sure Petcare Flap battery sensor."""
         self._id = _id
-        self._name = f"Flap {name.capitalize()} Battery Level"
+        self._sure_type = sure_type
+
         self._spc = spc
-        self._state = self._spc.states[SureThingID.FLAP.name][self._id].get("data")
+        self._spc_data: Dict[str, Any] = self._spc.states[self._sure_type].get(self._id)
+        self._state: Dict[str, Any] = {}
+
+        self._name = (
+            f"{self._sure_type.name.capitalize()} "
+            f"{self._spc_data['name'].capitalize()}"
+        )
 
         self._async_unsub_dispatcher_connect = None
 
     @property
-    def should_poll(self):
+    def name(self) -> str:
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def unique_id(self) -> str:
+        """Return an unique ID."""
+        return f"{self._spc_data['household_id']}-{self._id}"
+
+    @property
+    def available(self) -> bool:
+        """Return true if entity is available."""
+        return bool(self._state["online"])
+
+    @property
+    def is_on(self) -> Optional[bool]:
```

Comment:
Sensors don't support is_on. Sensors with binary values should be binary sensors.